File: 3931-process-string-with-special-operations-i/process-string-with-special-operations-i.java
class Solution {
    public String processStr(String s) {
        String res="";
        for(int i=0;i<s.length();i++)
        {
            char ch=s.charAt(i);
            if(ch=='*'){
                if(res!=""){
                     if (res == null || res.isEmpty()) 
                        res="";
                    else
                    res=res.substring(0,res.length()-1);
                }
            }
            else if(ch=='#')
                res+=res;
            else if(ch=='%'){
                String rev=new StringBuilder(res).reverse().toString();
                res=rev;
            }
            else
            res+=ch;

            }
            return res;
        }

    }

File: 3931-process-string-with-special-operations-i/process-string-with-special-operations-i.py
class Solution(object):
    def processStr(self, s):
        res=""
        for i in s:
            if i=='*':
                if res!="":
                    res=res[:-1]
                else:
                    res=""
            elif i=='#':
                res+=res
            elif i=='%':
                res=res[::-1]
            else:
                res+=i
        return res
